synapseclient/__main__.py

```diff
@@ -1871,8 +1871,10 @@ def _authenticate_login(
 
 def main():
     args = build_parser().parse_args()
-    synapseclient.USER_AGENT["User-Agent"] = (
-        "synapsecommandlineclient " + synapseclient.USER_AGENT["User-Agent"]
+    synapseclient.USER_AGENT[
```

Comment:
I might suggest a slightly different solution to create a second variable in Synapseclient instead of doing a string replacement. 
```
USER_AGENT_COMMAND_LINE = {
    "User-Agent": "synapsecommandlineclient/%s %s"
    % (__version__, requests.utils.default_user_agent())
}

```

Then use that here. The reason is that it us more clear what the agents are expected to be used by this library.

Reply:
This is the better solution. 👌

Reply:
Thanks for sharing.  Before adopting this solution,  I would like to confirm that this won't change the work of chain in the client. From what I've learned, the command line client wraps synapseclient functions but adds arg parser to it. There are other scripts utilizing synapseclient.USER_AGENT: 

1. [synapseclient/core/version_check.py](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/0b903e3797206650e0d9137516ce46eb1f454e77/synapseclient/core/version_check.py#L177). In this script, USER_AGENT is added to the headers of a API call to GitHub site and I checked the GitHub returns the content of the target site with or without the USER_AGENT. 

2. [synapseclient/client.py](https://github.com/Sage-Bionetworks/synapsePythonClient/blob/0b903e3797206650e0d9137516ce46eb1f454e77/synapseclient/client.py#L694). In line 694, the USER_AGENT is used to construct a request session. If I understand it correctly, this would trigger the downstream log of the USER_AGENT on Synapse server side.

Reply:
What the code is currently doing is redefining an attribute on the `USER_AGENT` constant defined in the client with a new value. This isn't the best, but it's fine because this code is only ever invoked from the CLI and temporary reassignments are just that - Temporary.

Ideally this would instead be a property on the Synapse class instance. But it would involve a larger number of changes.

What you would be doing with this suggestion is something like:

`synapseclient.USER_AGENT = synapseclient.USER_AGENT_COMMAND_LINE`

Reply:
Gotcha. On it.